src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -345,5 +369,46 @@ internal bool IsNetStandardFeatureFlagEnabled()
 
             return value == "1" || value == "true";
         }
+
+        internal IDictionary<int, bool> ParseHttpCodesToDictionary(string httpStatusErrorCodes)
+        {
+            string[] configurationsArray = string.Concat(httpStatusErrorCodes.Where(c => !char.IsWhiteSpace(c))).Split(',');
+
+            IDictionary<int, bool> httpErrorCodesDictionary = new Dictionary<int, bool>();
+
+            foreach (string statusConfiguration in configurationsArray)
+            {
+                int parsedStatus;
+
+                if (!Regex.IsMatch(statusConfiguration, @"^\d{3}-\d{3}$|^\d{3}$"))
+                {
+                    Log.Warning("Wrong format '{0}' for DD_HTTP_SERVER/CLIENT_ERROR_STATUSES configuration.", statusConfiguration);
+                }
+                else if (int.TryParse(statusConfiguration, out parsedStatus))
+                {
+                    httpErrorCodesDictionary[parsedStatus] = true;
+                }
+                else
+                {
+                    var statusRange = statusConfiguration.Split('-').Select(int.Parse).ToArray();
```

Comment:
I prefer to finish this line with the `Split` and Parse things in other lines. Using `Linq` + `Delegate to int.Parse` + `ToArray()` seems too much overhead for this simple operation.